xc/device.ml

```diff
@@ -1392,6 +1392,82 @@ module Vkbd = struct
 
 end
 
+module Vusb = struct
+  let vusb_controller_plug ~xs ~domid ~driver ~driver_id =
+    let is_running = Qemu.is_running ~xs domid in
+    match is_running with
+    | true  -> (
+      let open Qmp in
+      let qmp_cmd = Command (None, Device_add (driver, driver_id, None)) in
+      qmp_write domid qmp_cmd
+    )
+    | _ ->()
+
+  let vusb_plug ~xs ~domid ~id ~hostbus ~hostport ~version =
+    debug "Vusb plugged: vusb device plugged ";
+    let get_bus v =
+      if String.startswith "1" v then
+         "usb-bus.0"
+      else begin
+          (* Here plug usb controller according to the usb version*)
+          let usb_controller_driver = "usb-ehci" in
+          let driver_id = "ehci" in
+          vusb_controller_plug ~xs ~domid ~driver:usb_controller_driver ~driver_id;
+          "ehci.0"
```

Comment:
To avoid any problems when `driver_id` changes:
```
`Printf.sprintf "%s.0" driver_id
```

Reply:
updated. Thanks